### ISSUE TITLE
system/sef plugin background image parser regexp bug

### ISSUE THREAD

**Issue by nextend** [joomla/joomla-cms#7267]
## Bug
The problem is that I'm using images in the style attributes for background images and I'm set the image urls with a full url:

`<div style="background-image: url(&#039;http://test.org/test.png&#039;)"></div>`

Note that the ' in the url()  are properly encoded for HTML attributes. This is when the issue comes, the following regexp only match the not encoded ' and " characters and not the encoded ones. The regexp should be fixed by adding 
' => &#039;
" => &#034;

The regexp is in plugins/system/sef/sef.php which used to find image urls in style attributes to fix them to the current site's relative path:

$regex  = '#style\s*=\s*[\'\"](.*):\s*url\s*\([\'\"]?(?!/|' . $protocols . '|\#)([^\)\'\"]+)[\'\"]?\)#m';

## Test:
`<div style="background-image: url(&#039;http://test.org/test.png&#039;)"></div>`

### Result:
`<div style="background-image: url("/my/site/path/&#039;http://test.org/test.png&#039;")"></div>`

### Desired result: - (Regexp do not match this url.)
`<div style="background-image: url(&#039;http://test.org/test.png&#039;)"></div>`

**Comment by ggppdk** [comment]
The regular expression has some negative look ahead to avoid matching things like:
url( '/
url( 'http://

we could add   &#039;  to the negative lookahead:

Replacing:
```php
// Background image.
$regex  = '#style\s*=\s*[\'\"](.*):\s*url\s*\([\'\"]?(?!/|' . $protocols . '|\#)([^\)\'\"]+)[\'\"]?\)#m';
```

with:
```php
// Background image.
$regex  = '#style\s*=\s*[\'\"](.*):\s*url\s*\([\'\"]?(?!/|\&\#039;|' . $protocols . '|\#)([^\)\'\"]+)[\'\"]?\)#m';
```

**Comment by nextend** [comment]
For us the solution was to use uppercase URL in our HTML, as the SEF preg_match do not ignore case :)

<div style="background-image: URL(&#039;http://test.org/test.png&#039;)"></div>

Also I have one other use case when the regexp fails. Nowadays protocol relative urls are popular, so this would be perfect if you have a site with http and https too:
<div style="background-image: url('//test.org/test.png')"></div>

But the regular expression is not able to match the protocol so it fails and adds the previously mentioned path to the url.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7267">issues.joomla.org/joomla-cms/7267</a>.</sub>

**Comment by brianteeman** [comment]
Closed as we have a PR for testing #11412<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7267">issues.joomla.org/joomla-cms/7267</a>.</sub>
